onnx/defs/math/defs.cc

```diff
@@ -3412,19 +3438,47 @@ bool BuildContextDependentFunctionBodySCE(
       {{"axes"},
        "Constant",
        {},
-       {MakeAttribute("value", ToDimensionOneInt64Tensor(1))}});
+       {MakeAttribute("value", ToDimensionOneInt64Tensor(-1))}});
 
+  // Using stable implementation of LogSoftmax
+  // We need to unsqueeze the inputs on the last axis if they are 2D in order to make the 
+  // further Reshape and transpose valid.
```

Comment:
Just curious: is this really required? It seems like reshaping a 2D with target-shape {0,0,-1} should have the same effect, if I look at the reshape spec. Wonder if there is some bug/ambiguity in Reshape impl/spec.

Reply:
you are right, this unsqueeze can be skipped